api.d.ts

```diff
@@ -0,0 +1,647 @@
+interface Common {
+    config(new_config: any, new_value: any)
```

Comment:
Thanks for all the great work!
I appreciate to have this PR so I can skip a lot part of declaring custom type by myself!

Here is some feedback and suggestion that I found while using this type declaration, hope it helps.

I found that config function allows empty argument, which in case sets config if `CLOUDINARY_URL` is been set in environment.
I found configuration parameters in the doc, so added interface for that configuration

```ts
  // https://cloudinary.com/documentation/solution_overview#configuration_parameters
  interface Configuration {
    cloud_name: string
    api_key: string
    api_secret: string
    secure?: boolean
    upload_preset?: string
    cdn_subdomain?: boolean
    private_cdn?: boolean
    cname?: boolean
    secure_distribution?: string
    static_image_support?: boolean
    enhance_image_tag?: boolean
  }

  interface Common {
    config(): Configuration
    config(new_config: keyof Configuration): any
    config(new_config: Configuration): Configuration
    config(new_config: keyof Configuration, new_value: string | boolean): Configuration
```